openstack/sharedfilesystems/apiversions/errors.go

```diff
@@ -0,0 +1,23 @@
+package apiversions
+
+import (
+	"fmt"
+)
+
+// ErrVersionNotFound is the error when the requested API version
+// could not be found.
+type ErrVersionNotFound struct{}
+
+func (e ErrVersionNotFound) Error() string {
```

Comment:
I agree that there is no existing error that would fit.

My point is that `BaseError` and `func choseErrString` are not used. However, I understand that you don't see much benefit using `BaseError` and `func choseErrString`.